evml/keras/models.py

```diff
@@ -136,8 +132,52 @@ def build_neural_network(self, inputs, outputs):
                 nn_model = GaussianNoise(self.noise_sd, name=f"ganoise_h_{h:02d}")(
                     nn_model
                 )
-        nn_model = Dense(outputs, name="dense_last")(nn_model)
+
+        if last_layer == "Dense":
+            nn_model = Dense(outputs, name="dense_last")(nn_model)
+        elif last_layer == "DenseNormal":
+            nn_model = DenseNormal(outputs, name="DenseNormal", eps=self.eps)(nn_model)
+        elif last_layer == "DenseNormalGamma":
+            nn_model = DenseNormalGamma(outputs, name="DenseNormalGamma", eps=self.eps)(
+                nn_model
+            )
+        else:
+            raise ValueError("Invalid last_layer type. Use 'Dense', 'DenseNormal', or 'DenseNormalGamma'.")
+
         self.model = Model(nn_input, nn_model)
+
+        if self.optimizer == "adam":
+            self.optimizer_obj = Adam(learning_rate=self.lr)
+        elif self.optimizer == "sgd":
+            self.optimizer_obj = SGD(learning_rate=self.lr, momentum=self.sgd_momentum)
+
+        if self.metrics == "mae":
+            metrics = self.mae
+        elif self.metrics == "mse":
+            metrics = self.mse
+        else:
+            metrics = None
+            
+        self.model.compile(
+            optimizer=self.optimizer_obj,
+            loss=self.loss,
+            loss_weights=self.loss_weights,
+            metrics=metrics,
```

Comment:
The `loss_weights` argument here is not the same as `class_weights` in the `.fit()` method which I think is your intention.

Reply:
I will test again to see if this actually works (despite keras saying it does); if not it will be removed. But I will add class_weights as an option for when the model is a standard classifier and not evidential (we could enable the experimental class weights I put into the loss, but I cannot confirm that is really correct ...)